agents/xenapi/fence_xenapi.py

```diff
@@ -127,9 +127,14 @@ def connect_and_login(options):
 	username = options["--username"]
 	password = options["--password"]
 
+	# Allow bypassing SSL verification for hosts without SSL certificates
+	skipVerification = False
+	if "--ssl-insecure" in options:
+		skipVerification = True
+	
 	try:
 		# Create the XML RPC session to the specified URL.
-		session = XenAPI.Session(url)
+		session = XenAPI.Session(url, None, None, 0, 1, skipVerification)
```

Comment:
Session() doesnt appear to have a parameter for skipVerification:
https://github.com/ClusterLabs/fence-agents/blob/main/lib/XenAPI.py.py#L121

Reply:
Looks like the XenAPI library bundled in this repo is significantly outdated. The latest version from https://github.com/xapi-project/xen-api/blob/master/scripts/examples/python/XenAPI/XenAPI.py should be loaded instead.